### PR TITLE
add update instructions to retryer readme

### PR DESCRIPTION
**Pull request by TrevorFarrelly**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

### CODE REVIEW

File: tools/flaky-test-retryer/README.md [hunk 1]
```diff
@@ -97,6 +97,14 @@ automatic retry.
 
 if all retries are expended without a success.
 
+## Updating
+
+1. Run `make push_versioned` in /images/flaky-test-retryer/
```
Comment (adrcunha) [inline tools/flaky-test-retryer/README.md]:
```suggestion
1. Run `make push_versioned` in `/images/flaky-test-retryer`.
```

File: tools/flaky-test-retryer/README.md [hunk 2]
```diff
@@ -97,6 +97,14 @@ automatic retry.
...
+2. Note the image tag, in the format `v[date]-[hash]`.
+3. Replace the current image tag in [retryer_service.yaml](https://github.com/knative/test-infra/blob/master/tools/flaky-test-retryer/gke_deployment/retryer_service.yaml#L40)
+with the tag generated by make, excluding `-dirty` suffix if it exists.
```
Comment (adrcunha) [inline tools/flaky-test-retryer/README.md]: You should not drop `-dirty`, as it's really part of the flag. Actually, you shouldn't encourage anyone of using a "dirty" image, as it's untrackable.

File: tools/flaky-test-retryer/README.md (hunk 1 above)
Comment (TrevorFarrelly) [reply tools/flaky-test-retryer/README.md]: fixed

File: tools/flaky-test-retryer/README.md (hunk 2 above)
Comment (TrevorFarrelly) [reply tools/flaky-test-retryer/README.md]: fixed, didn't understand what -dirty meant at first. Should be good now.

File: tools/flaky-test-retryer/README.md [hunk 3]
```diff
@@ -97,6 +97,14 @@ automatic retry.
 
 if all retries are expended without a success.
 
+## Updating
+
+1. Run `make push_versioned` in `/images/flaky-test-retryer/` with a clean working tree.
+2. Note the image tag, in the format `v[date]-[hash]`.
+3. Replace the current image tag in [retryer_service.yaml](https://github.com/knative/test-infra/blob/master/tools/flaky-test-retryer/gke_deployment/retryer_service.yaml#L40)
```
Comment (adrcunha) [inline tools/flaky-test-retryer/README.md]: Use a permalink, so the link won't point to the wrong entry if this file changes.

```suggestion
3. Replace the current image tag in [retryer_service.yaml](https://github.com/knative/test-infra/blob/751583ee73a463cf6449817afda38d63b29dfff7/tools/flaky-test-retryer/gke_deployment/retryer_service.yaml#L40)
```

**Comment by adrcunha** [comment]
One last thing.

/approve

**Pull request merged**